src/main/java/uk/gov/pay/ledger/transaction/search/common/TransactionSearchParams.java

```diff
@@ -135,33 +141,29 @@ public String generateQuery() {
         if (isNotBlank(cardHolderName)) {
             sb.append(" AND t.cardholder_name ILIKE :" + CARDHOLDER_NAME_FIELD);
         }
-
         if (fromDate != null) {
             sb.append(" AND t.created_date > :" + FROM_DATE_FIELD);
         }
         if (toDate != null) {
             sb.append(" AND t.created_date < :" + TO_DATE_FIELD);
         }
         if (paymentStates != null && !paymentStates.isEmpty()) {
-
-            if (paymentStates.has("CREATED")) {
-                sb.append(" AND t.state in ('CREATED')");
-            } else {
-                sb.append(" AND false");
-            }
+            //TODO implement
+        }
+        if (isNotBlank(state)) {
+            sb.append(" AND t.state = :" + STATE_FIELD);
         }
-
         if (refundStates != null) {
-            sb.append(" AND false");
+            //TODO implement
         }
         if (cardBrands != null && !cardBrands.isEmpty()) {
-            sb.append(" AND false");
+            sb.append(" AND t.card_brand IN(" + inClause(cardBrands) + ")");
```

Comment:
inClause generated manually (not by binding) is the worrying thing.

It seems it's theoretically possible to bind the list instead of manually generating string (http://jdbi.org/#_binding_arguments), but I'll have a look at it in separate PR (this one is already to big)